### PR TITLE
introduce RAG pipeline via Ollama & PineCone

### PR DESCRIPTION
**Pull request by thisthat**

### CODE REVIEW

File: deployment/travel-advisor.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+              readOnly: true
+              mountPath: "/etc/secrets"
+      volumes:
+        - name: secrets
+          projected:
+            sources:
+              - secret:
+                  name: dynatrace
+              - secret:
+                  name: pinecone
```
Comment (thisthat) [inline deployment/travel-advisor.yaml:49]: to make it work with codespace, we would need two K8s secrets in the ns `travel-advisor` :
1. `dynatrace` with the DT API Key in a `token` field
2. `pinecone` with the PineCone key in a `api-key` field

File: deployment/travel-advisor.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      containers:
+        - name: travel-advisor
+          image: thisthatdc/travel-advisor:v0.1.3
+          ports:
+            - name: http
+              containerPort: 8080
+              protocol: TCP
+          env:
+            - name: OTEL_ENDPOINT
+              value: "https://xbw95514.dev.dynatracelabs.com/api/v2/otlp"
```
Comment (thisthat) [inline deployment/travel-advisor.yaml:32]: also here, this should be generated based on the DT tenant used

**Comment by agardnerIT** [comment]
1. Will this PR replace what we have existing or add a new deployment option to the 2 we already have?
2. Does this still use 100% OTEL or does it introduce a dependency on the OneAgent
3. This PR makes everything self-contained and removes the dependency on OpenAI, right?
4. What is the impact on codespace startup time for keeping everything self contained? I imagine Ollama downloading the model will take a long time
5. In any case, the `docs` need to be updated and redeployed

**Comment by thisthat** [comment]
> Will this PR replace what we have existing or add a new deployment option to the 2 we already have?
> What is the impact on codespace startup time for keeping everything self contained? I imagine Ollama downloading the model will take a long time

I would suggest that it can live next to the existing RAG. If you wanna have everything self-contained (which can take longer time), then you have this. Otherwise, you can use the OpenAI version. 

> Does this still use 100% OTEL or does it introduce a dependency on the OneAgent

This is pure OTel, no OA is required. 


> This PR makes everything self-contained and removes the dependency on OpenAI, right?

Yes

> In any case, the docs need to be updated and redeployed

I suggest doing this in a follow up since I would see this as an addition than a replacement

**Pull request merged**